src/ex6/client/src/components/TodosModal.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect, useCallback } from "react";
+import Modal from "react-modal";
+import { getTodos } from "../todos-client";
+import TodoItem from "./TodoItem";
+import TodosHeader from "./TodosHeader";
+
+export default function RegistarModal(props) {
+  Modal.setAppElement("#portal");
+
+  const [todos, setTodos] = useState();
+
+  const fetchTodos = useCallback(async () => {
+    const todos = await getTodos();
+    setTodos(todos);
+  }, []);
+
+  useEffect(() => {
+    fetchTodos();
+  }, [fetchTodos]);
+
+  let displayTodos = "loading";
+  if (todos) {
+    displayTodos = todos.map((item) => {
+      return (
+        <TodoItem
+          key={item.id}
+          todoId={item.id}
+          todoName={item.itemName}
+          todoStatus={item.status}
+          onItemChange={fetchTodos}
+        />
+      );
+    });
+  }
```

Comment:
```suggestion
const displayTodos = useMemo(async () => {
  if (!todos) return "loading";
    displayTodos = todos.map((item) => {
      return (
        <TodoItem
          key={item.id}
          todoId={item.id}
          todoName={item.itemName}
          todoStatus={item.status}
          onItemChange={fetchTodos}
        />
      );
    });
  }, [todos])
```

Reply:
a common practice is to add a loading state which is boolean. set to false after you finished the fetch
(in case the fetch can return null when its empty, you can be stuck in loading state, this way you do not rely on the result but on a flag that indicates your state)